src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1951,12 +1951,21 @@ internal static SpecialType GetEnumPromotedType(SpecialType underlyingType)
                 case BinaryOperatorKind.IntRightShift:
                 case BinaryOperatorKind.NIntRightShift:
                     return valueLeft.Int32Value >> valueRight.Int32Value;
+                case BinaryOperatorKind.IntUnsignedRightShift:
+                    return (int)(((uint)valueLeft.Int32Value) >> valueRight.Int32Value); // Switch to `valueLeft.Int32Value >>> valueRight.Int32Value` once >>> becomes available
+                case BinaryOperatorKind.NIntUnsignedRightShift:
+                    return (valueLeft.Int32Value >= 0) ? valueLeft.Int32Value >> valueRight.Int32Value : null;
```

Comment:
I didn't understand why we can't compute the constant result when `valueLeft.Int32Value < 0`

Reply:
> I didn't understand why we can't compute the constant result when ```valueLeft.Int32Value < 0```

Binary representation of negative values depends on the actual size of the type.

Reply:
Worked it out on paper. Makes sense